### PR TITLE
RI: CherryPick of 1.2.1 bug fixes from master

### PR DESCRIPTION
**Pull request by shalinijoshi19**
AdaptiveCards 1.2.1 payload from master. Contains commits from the following 1.2.1 PRs
https://github.com/microsoft/AdaptiveCards/pulls?utf8=%E2%9C%93&q=is%3Apr+label%3APotential-1.2.1

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/3278)

### CODE REVIEW

**Comment by shalinijoshi19** [comment]
baah! this needs to be against v.1.2.0 not master! Resubmitting

File: source/uwp/AdaptiveCardTestApp/Expected/v1.0.Elements.Input.ToggleToJson.json [hunk 1]
```diff
@@ -1 +1,5 @@
-{"actions":[{"title":"OK","type":"Action.Submit"}],"body":[{"text":"Toggle Input","type":"TextBlock"},{"id":"acceptTerms","isRequired":false,"title":"I accept the terms and agreements","type":"Input.Toggle","value":"true","valueOff":"false","valueOn":"true"}],"type":"AdaptiveCard","version":"1.0"}
\ No newline at end of file
+<<<<<<< HEAD
```
Comment (almedina-ms) [inline source/uwp/AdaptiveCardTestApp/Expected/v1.0.Elements.Input.ToggleToJson.json]: This file didn't merge correctly

**Comment by shalinijoshi19** [comment]
> baah! this needs to be against v.1.2.0 not master! Resubmitting

Ok this is done.. no ability to create PR's against tags but at we had a release/1.2 branch that should help fix the PR up

File: source/uwp/AdaptiveCardTestApp/Expected/v1.0.Elements.Input.ToggleToJson.json (hunk 1 above)
Comment (shalinijoshi19) [reply source/uwp/AdaptiveCardTestApp/Expected/v1.0.Elements.Input.ToggleToJson.json]: Tx

---
In reply to: [308932979](https://github.com/microsoft/AdaptiveCards/pull/3278#discussion_r308932979) [](ancestors = 308932979)

**Comment by paulcam206** [comment]
Looks like one of the iOS tests failed:

```
ADCIOSVisualizerTests
    ✗ testACRTextView, ((cardParseResult && cardParseResult.isValid) is true) failed
```

**Comment by shalinijoshi19** [comment]
> Looks like one of the iOS tests failed:
> 
> ```
> ADCIOSVisualizerTests
>     ✗ testACRTextView, ((cardParseResult && cardParseResult.isValid) is true) failed
> ```

Yes I asked @nesalang to help look into the failure and fix up any bad merge there ! @nesalang feel free to pull down the branch, make the changes and push them up directly! Thanks again!

**Comment by paulcam206** [comment]
looks good overall! :)

**Comment by almedina-ms** [comment]
I checked the android and .net files and they look good

**Pull request merged**